pkg/apis/core/validation/utils.go

```diff
@@ -191,11 +192,91 @@ func validateKubernetesVersions(versions []core.ExpirableVersion, fldPath *field
+		}
+
+		if duplicateLifecycle(version.Lifecycle) {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("lifecycle"), version, fmt.Sprintf("Invalid lifecycle of %s: duplicate classification in lifecycle.", version.Version)))
+		}
+		if lifecycleInOrder(version.Lifecycle) {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("lifecycle"), version, fmt.Sprintf("Invalid lifecycle of %s: lifecycle classifications not in order, must be preview -> supported -> deprecated -> expired.", version.Version)))
+		}
+
+		if lifecycleStartTimesInOrder(version.Lifecycle) {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("lifecycle"), version, fmt.Sprintf("Invalid lifecycle of %s: lifecycle classification startTimes not in order, dates must be preview -> supported -> deprecated -> expired.", version.Version)))
+		}
 	}
 
 	return allErrs
 }
 
+// duplicateLifecycle checks if there are any duplicate entries in the provided
+// lifecycle slice and returns a boolean value indicating whether duplicates were found in the lifecycle slice.
+func duplicateLifecycle(lifecycle []core.ClassificationLifecycle) bool {
+	seen := make(map[core.ClassificationLifecycle]bool)
+	for _, value := range lifecycle {
+		if seen[value] {
+			return true
+		}
+		seen[value] = true
+	}
+	return false
+}
+
+// lifecycleInOrder checks if the provided lifecycle slice is in  the expected order.
+// The order is not required for functionality but should ensure better readability.
+func lifecycleInOrder(lifecycle []core.ClassificationLifecycle) bool {
+	if len(lifecycle) <= 1 {
+		return true
+	}
+
+	order := map[core.VersionClassification]int{
+		core.ClassificationPreview:    0,
+		core.ClassificationSupported:  1,
+		core.ClassificationDeprecated: 2,
+		core.ClassificationExpired:    3,
+	}
+
+	for i, curr := range lifecycle[:len(lifecycle)-1] {
+		currVal, existsCurr := order[curr.Classification]
+		nextVal, existsNext := order[lifecycle[i+1].Classification]
+
+		if !existsCurr || !existsNext {
+			return false
+		}
+		if currVal >= nextVal {
+			return false
+		}
+	}
+
+	return true
+}
+
+// lifecycleStartTimesInOrder checks if the provided lifecycle slice has start times in order.
+// It does not ensure the correct order of the classifications but if the elements in the
+// list have dates after each other. The order must be tested via `lifecycleInOrder`.
+func lifecycleStartTimesInOrder(lifecycle []core.ClassificationLifecycle) bool {
+	if len(lifecycle) <= 1 {
+		return true
+	}
+
+	for i := 0; i < (len(lifecycle) - 2); i++ {
+		var startTime time.Time
+		if lifecycle[i].StartTime == nil {
+			startTime = time.Unix(0, 0)
+		} else {
+			startTime = lifecycle[i].StartTime.UTC()
+		}
+
+		if startTime.After(lifecycle[i+1].StartTime.UTC()) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
This would make it similar to the approve above:

```golang
// lifecycleStartTimesInOrder checks if the provided lifecycle slice has start times in order.
// It does not ensure the correct order of the classifications but if the elements in the
// list have dates after each other. The order must be tested via `lifecycleInOrder`.
func lifecycleStartTimesInOrder(lifecycle []core.ClassificationLifecycle) error {
	var previousStartTime time.Time

	for i, l := range lifecycle {
		if i == 0 {
			if l.StartTime == nil {
				l.StartTime = &v1.Time{}
			}

			previousStartTime = l.StartTime.Time

			continue
		}

		if l.StartTime == nil {
			return fmt.Errorf("only the first lifecycle element can have the start time optional")
		}

		currentStartTime := l.StartTime.Time

		if currentStartTime.After(previousStartTime) {
			return fmt.Errorf("lifecycle start times must be monotonically increasing")
		}

		previousStartTime = currentStartTime
	}

	return nil
}
```